src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -2326,51 +2329,22 @@ namespace WaterThermalTanks {
             Tank.IgnitionDelay = DataIPShortCuts::rNumericArgs(7); // Not yet implemented
 
             // Validate Heater Fuel Type
-            {
+            bool FuelTypeFirstPass(false);
+            UtilityRoutines::ValidateFuelType(DataIPShortCuts::cAlphaArgs(4), Tank.FuelType, FuelTypeFirstPass);
+            if (FuelTypeFirstPass) {
```

Comment:
Why is this `FuelTypeFirstPass` needed?

Reply:
This is because the common ValidateFuelType function (https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/UtilityRoutines.cc#L474-L522) covers 10 types of fuel (Electricity, Naturalgas, gasoline, propane… etc) according to GetInput of components like boiler, chiller, etc. For this case in WaterHeater, it has two more additional fueltypes; steam and district heating. Once I set FuelTyepFirstPass here rather than changing the common function to have steam and district heating because it should be an error if "steam" or "district heating" is an input of components like boiler, chiller, etc. Please let me know if you have another idea.

Reply:
I see. So, another followup issue (which will likely result in IDD changes): Steam is an odd resource type. I see it show up in just a few places in the IDD. LifeCycleCost:UsePriceEscalation/UseAdjustement have Steam, but not DistrictHeating and DistrictCooling - seems they should?

OtherEquipment also allows steam and districtheating/cooling - so what's it doing? Oh, it's using ExteriorEnergyUse::ValidateFuelType - which includes steam and districtheating/cooling. So, there's still a duplicate function out there - but that one is returning an enum (which is better). So, ultimately, that function should probably move to UtilityFunctions and everywhere should convert over to use the enums, but that can be another pass.

In general, however, since only certain equipment types allow steam and district*, how about adding those `UtilityRoutines::ValidateFuelType` and adding an optional argument like `AllowSteamAndDistrict` that defaults to false?

Reply:
And related to above - should the ABUPS report have a column for steam? If a water heater uses fuel type = steam, does the energy use show up in ABUPS?

Reply:
Seems like steam is included in district heating in ABUPS report, no separated column for steam. District heating column shows the sum of purchased heating and steam. (https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/OutputReportTabular.cc#L7652) 
Should we have a separate column for steam? @mjwitte @Myoldmopar

Reply:
@dareumnam  Let's not change that now. That's something we can consider after release. To be honest, I don't see Steam being any different from DistrictHeating, but it's been around a long time.

Also, let's not worry about collapsing the functions again for this release. That can happen later as time allows. This is good progress here.